### DISCUSSION TITLE
Does pytest automatically add the project root directory to `sys.path` when a package is installed in the editable mode?

### DISCUSSION THREAD

**Discussion by harupy** [pytest-dev/pytest#12530] [Q&A]
Let's say I have the following project structure:

```
pyproject.toml
src/
    mypkg/
        __init__.py
        app.py
        view.py
tests/
    test_app.py
    test_view.py
    ...
```

Does `pytest` automatically detect the project root directory and add it to `sys.path` when a package is installed in the editable mode?

```
pip install -e .
pytest tests   # Adds root to sys.path?
```

**Comment by RonnyPfannschmidt** [comment]
That depends on whether a `__init__.py` exist in test 

If it does,the root is added, else tests is

**Reply by harupy** [reply]
@RonnyPfannschmidt Thanks for the comment! I was investigating why `sys.path` has the project root when the package is installed in the editable model. It turns out python appends it. pytest test is unrelated. Closing this discussion.

```
pip install -e .
pytest  # sys.path has the project root

# vs.

pip install .
pytest  # sys.path doesn't
```

**Reply by RonnyPfannschmidt** [reply]
If your python package uses a src layout, I would expect only src on the python path

I recommend investigating whether there's a packaging mistake
